tools/pcode-lifter/main.cpp

```diff
@@ -152,22 +154,29 @@ namespace {
         int bit_size = std::stoi(lang_vec[2]);
         auto is_le   = (lang_vec[1] == "LE");
 
-        if (arch == "X86" || arch == "X86-64") {
+        auto is_equal = [&](std::string a, std::string b) -> bool {
```

Comment:
this is a bit minor, but can't we just transform the source string once via something like:
```
std::transform(str.begin(), str.end(), str.begin(),
               [](unsigned char c){ return std::toupper(c); });
```

and then do normal operator == ?

Reply:
yes we can do that as well, transform both the string and then use `operator ==`.